frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,315 @@
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { match } from 'react-router';
+
+import { Grid, GridItem, ActionGroup, Button, Alert } from '@patternfly/react-core';
+
+import {
+  LoadingBox,
+  ListDropdown,
+  ButtonBar,
+  history,
+  ResourceIcon,
+  resourceObjPath,
+  HandlePromiseProps,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import {
+  K8sKind,
+  referenceForModel,
+  K8sResourceKind,
+  k8sCreate,
+  referenceFor,
+  VolumeSnapshotClassKind,
+  StorageClassResourceKind,
+  PersistentVolumeClaimKind,
+  k8sGet,
+} from '@console/internal/module/k8s';
+import { connectToPlural } from '@console/internal/kinds';
+import {
+  PersistentVolumeClaimModel,
+  VolumeSnapshotModel,
+  VolumeSnapshotClassModel,
+  StorageClassModel,
+  NamespaceModel,
+} from '@console/internal/models';
+import { PVCDropdown } from '@console/internal/components/utils/pvc-dropdown';
+import { apiVersionForModel } from '@console/kubevirt-plugin/integration-tests/tests/utils/selectors';
+import { getName, getNamespace } from '@console/shared';
+import { PVCStatus } from '@console/internal/components/persistent-volume-claim';
+
+import './_create-volume-snapshot.scss';
+
+const SnapshotClassDropdown: React.FC<SnapshotClassDropdownProps> = (props) => {
+  const { selectedKey, pvcSC } = props;
+  const kind = referenceForModel(VolumeSnapshotClassModel);
+  const resources = [{ kind }];
+  const [scObj, setSCObj] = React.useState<StorageClassResourceKind>(null);
+
+  React.useEffect(() => {
+    k8sGet(StorageClassModel, pvcSC)
+      .then((sc) => setSCObj(sc))
+      .catch((error) =>
+        // eslint-disable-next-line no-console
+        console.error(error),
+      );
+  }, [pvcSC]);
+
+  const filter = (snapshotClass: VolumeSnapshotClassKind) => {
+    let flag: boolean;
+    !scObj || !scObj.provisioner
+      ? (flag = false)
+      : (flag = scObj?.provisioner.includes(snapshotClass?.driver));
+    return flag;
+  };
+
+  return (
+    <ListDropdown
+      {...props}
+      desc="Volume Snapshot Class with same provisioner as claim"
+      dataFilter={filter}
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select snapshot class"
+      selectedKey={selectedKey}
+    />
+  );
+};
+
+const PVCSummary: React.FC<PVCSummaryProps> = ({ persistentVolumeClaim }) => {
+  const storageClass = persistentVolumeClaim?.spec?.storageClassName;
+  const requestedCapacity = persistentVolumeClaim?.spec?.resources?.requests?.storage;
+  const accessModes =
+    persistentVolumeClaim?.spec?.accessModes.reduce(
+      (allModes: string, mode: string) => `${mode} ${allModes}`,
```

Comment:
```suggestion
      (allModes, mode) => `${mode} ${allModes}`,
```
Implicit from the initial value and PersistentVolumeClaimKind